packages/firebase_vertexai/firebase_vertexai/test/imagen_test.dart

```diff
@@ -0,0 +1,241 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:firebase_vertexai/src/error.dart';
+import 'package:firebase_vertexai/src/imagen_content.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  group('ImagenInlineImage', () {
+    test('fromJson with valid base64', () {
```

Comment:
I think in future we should think about using the latest vertexai mock response from here: https://github.com/FirebaseExtended/vertexai-sdk-test-data

It's used in the firebase-js-sdk and now react-native-firebase. It allows for tests much closer to e2e:

See equivalent in firebase-js-sdk: https://github.com/firebase/firebase-js-sdk/pull/8683/files#diff-395c4eb9cd7bf5d84093b3d85f102bf133b761b997cb2177d5bc5e254527f4baR75-R159

Appreciate this is a much bigger task so hence why I won't allow it to block, it requires a script to pull in the mock responses and additional setup: https://github.com/firebase/firebase-js-sdk/blob/main/scripts/update_vertexai_responses.sh

Reply:
Good suggestion, will work on it.